packages/lesswrong/server/resolvers/anthropicResolvers.ts

```diff
@@ -119,20 +121,49 @@ async function getQueryContextDecision(args: BasePromptArgs): Promise<RagContext
   return parsedResponse.strategy_choice;
 }
 
+async function getProvidedPosts(query: string, context: ResolverContext): Promise<PostsPage[]> {
+  const postIdRegex = /\/([a-zA-Z0-9]{17})(?=[/#?&)]|$)/g;
+  const postIdMatches = [];
+  let match;
+  while ((match = postIdRegex.exec(query)) !== null) {
+    postIdMatches.push(match[1]);
+  }
+
+  const postIds = uniq(postIdMatches);
+  const posts = await getPostsWithContents(postIds, context);
+
+  return posts;
+}
+
 // TODO: come back and refactor the query context decision to actually use the embedding distance results (including the post titles)
-async function getContextualPosts({ content: query, currentPost, postContext, context }: GetContextMessageArgs): Promise<PostsPage[]> {
-  const contextSelectionCode = await getQueryContextDecision({ query, postContext, currentPost });
+async function getContextualPosts({ content: query, ragMode, currentPost, postContext, context }: GetContextMessageArgs): Promise<LlmPost[]> {
+
+  const ragModeMapping = {
+    'CurrentPost': 'current-post-only',
+    'Search': 'both',
+    'None': 'none',
+    // 'Recommendation': 'recommendation',
+    'Provided': 'provided',
+    // 'Rationality Tutor': 'rationality-tutor',
+  }
```

Comment:
This kind of thing should live outside of the function, since it's a totally static const (and it needs to be cleaned up)

Reply:
comments intentionally left in as examples